dklearn/grid_search.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import absolute_import, print_function, division
+
+import numpy as np
+from dask import threaded
+from dask.base import tokenize
+from dask.context import _globals
+from dask.delayed import Delayed
+from sklearn.base import is_classifier, BaseEstimator
+from sklearn.cross_validation import check_cv
+from sklearn.grid_search import (_CVScoreTuple, _check_param_grid,
+                                 ParameterGrid, ParameterSampler)
+from sklearn.metrics.scorer import check_scoring
+from sklearn.utils import indexable, safe_indexing
+
+from .core import from_sklearn, unpack_arguments
+
+
+def _fit_and_score(estimator, X_name, y_name, scorer, train, test,
+                   parameters, fit_params):
+    estimator = estimator.set_params(**parameters)
+    n_samples = len(test)
+
+    # Extract train and test data
+    token = tokenize(X_name, y_name, train, test)
+    X_train_name = 'X_train-' + token
+    y_train_name = 'y_train-' + token
+    X_test_name = 'X_test-' + token
+    y_test_name = 'y_test-' + token
+
+    train, test, dsk = unpack_arguments(train, test)
+    dsk[X_train_name] = (safe_indexing, X_name, train)
+    dsk[y_train_name] = (safe_indexing, y_name, train)
+    dsk[X_test_name] = (safe_indexing, X_name, test)
+    dsk[y_test_name] = (safe_indexing, y_name, test)
```

Comment:
A mix of both of those. Biggest one is avoiding repeating `tokenize` calls on parameters. These can add up.